### ISSUE TITLE
common:/LibCore/v30/bslCore.brs path improperly modified in component libraries

### ISSUE THREAD

**Issue by triwav** [rokucommunity/roku-debug#106]
If you include it like `<script type="text/brightscript" uri="common:/LibCore/v30/bslCore.brs"/>` in a component library it tries to convert that to something like `<script type="text/brightscript" uri="common:/LibCore/v30/bslCore__lib2.brs"/>` which then causes the component library to fail to load due that file not existing. Probably the best way to handle is to make sure that the path is `pkg:/` before trying to modify the path for it.

**Comment by TwitchBronBron** [comment]
Here's a unit test that can be placed within `ProjectManager.spec.ts` somewhere.

```typescript
describe('postfixFiles', () => {
    it('skips "common:/" paths', async () => {
        let project = new ComponentLibraryProject(params);
        project.fileMappings = [];
        fsExtra.outputFileSync(`${params.stagingFolderPath}/source/main.brs`, '');
        fsExtra.outputFileSync(`${params.stagingFolderPath}/components/Component1.xml`, `
            <component name="CustomComponent" extends="Rectangle">
                <script type="text/brightscript" uri="common:/LibCore/v30/bslCore.brs"/>
                <script type="text/brightscript" uri="CustomComponent.brs"/>
                <script type="text/brightscript" uri="pkg:/source/utils.brs"/>
            </component>
        `);
        await project.postfixFiles();
        expect(
            fsExtra.readFileSync(`${params.stagingFolderPath}/components/Component1.xml`).toString()
        ).to.eql(`
            <component name="CustomComponent" extends="Rectangle">
                <script type="text/brightscript" uri="common:/LibCore/v30/bslCore.brs"/>
                <script type="text/brightscript" uri="CustomComponent__lib0.brs"/>
                <script type="text/brightscript" uri="pkg:/source/utils__lib0.brs"/>
            </component>
        `);
    });
});
```

**Comment by philandersonaccedo** [comment]
Picking up this ticket for investigation

**Comment by philandersonaccedo** [comment]
Notes on 1-1 session

- Issues with XML library import eg <script type="text/brightscript" uri="common:/LibCore/v30/bslCore.brs" /> 
- gets renamed incorrectly 
- so goal will be to exclude any files beginning with common:/ from file renaming scripts\
- follow instructions in README at https://github.com/rokucommunity/sample-projects/tree/master/component-library-and-project to get the project up and running remembering to 
- add the HOST IP in the MANIFEST
- observe 'out' folder and see
-   > .roku-deploy-staging APP
-   > component-libraries/complib.zip
- .... and both combine to make ... 
-   > roku-deploy.zip which gets deployed to the TV and run as the Roku app ...
- ... we notice in out/component-libraries/complib/components/RedditViewer.xml line 3 we have a renamed library import as   <script type="text/brightscript" uri="RedditViewer__lib0.brs" />. .... the suffix has been added so that we can identify the component library if a) there is an error or b) the file name get truncated ...


Background
============

- Talking about importing an external component library into your app ...
- Can be used for example to publish a 'thin' app to the Roku store and the 'guts' of the app are imported at runtime into the app ...
- Worked example on Roku Github Samples
  - https://github.com/rokucommunity/sample-projects
    - https://github.com/rokucommunity/sample-projects/tree/master/component-library-and-project
    -   when we attempt to run and debug this project we see that the imported library URL is failing because it's now renamed
    - from common:/LibCore/v30/bslCore.brs
    - to      common:/LibCore/v30/bslCore__lib0.brs
    - ie it has ___lib0 appended to the file name and for a 'common' library this should not be happening ...


Setting Up The Environment For The Fix
================================
    - we have a session running Roku-Debug
    - we can run a script to auto-run these 
    - switch to vscode-brightscript-language master branch and run
    - npm run install-local which runs a script in scripts/install-local.js
    -   which installs files for 
    -       ./roku-debug
    -       ./roku-deploy
    -       ./brighterscript
    -        ./brighterscript-formatter
    -  then runs a global
    -           npm install
    - so that's 5 script installs in total ... all done from one command ...
    -   ...... when all is done (and can run several times to ensure all libraries are in sync ... when all is done we can see some changes to package.json and package-lock.json ...

Task Watch

- Now run Command-P 'task watch' and select 'roku-debug' which starts a watcher watching for file changes ... 
- Now in VSCode click 'Run And Debug' and choose 'Extension and Debug Server' to launch a new instance of VSCode running the debug server
- Using the new instance of VSCode, called 'Extension Development Host', use it to now run the previous project we were running, namely 'sample-projects/component-library-and-project'
-  ..... before we run the project though, must enter a new line in 'launch.json'
                         "debugServer": 4711,   after "name":"Debug"

In VSCode 'workspace' ie vscode-brightscript-language now open up 'roku-debug/src/managers/ProjectManager.ts' which will be the right point to start our work.  Can add a break point to stop the code and check ... eg line 34 this.breakpointManager = breakpointManager;

- Now back in Extension+Debug Server, run with F5 and the breakpoint should hit.
- Continue on the breakpoint ... code should now run ... but crash is happening with illegal renaming of common library adding the suffix which it should not ...




Summary
==========

1. vscode-brightscript-language
2. npm run install-local
3. add a breakpoint at 'roku-debug/src/managers/ProjectManager.ts' at line 34
4. Command-P 'task watch' select 'roku-debug' which starts watch listener ...
5. Run with 'Run And Debug' option choosing 'Extension And Debug Server' which launches new instance of VSCode running as 'Extension Development Host' 
6. Using 'Extension Development Host' add a new line to 'launch.json'. ...... debugServer:4711 
7. open and run project 'sample-projects/component-library-and-project'
8. add in the extra line with the common:/ library mention to see and debug the crash ...    <script type="text/brightscript" uri="common:/LibCore/v30/bslCore.brs"/>

**Comment by philandersonaccedo** [comment]
Did about 1-2 hours of digging; can now clearly see the whole work flow and how everything is pieced together - started to work in detail on the file renaming aspect now also ... so good progress :)

**Comment by philanderson888** [comment]
Have identified the point in the code where the path renaming is taking place, just working through the fix ...

**Comment by philanderson888** [comment]
Also seems like the 'Task: Run ... watch' command is now being automatically implemented also

**Comment by philanderson888** [comment]
After a lot of digging, finally got a hold of the 'big picture' of what the code is doing, all of the moving parts, and was able to drill down into the code in order to locate the specific part in the code where the file name alteration is being done, and prevent it if this is a library file.

Added test also and ran it; seems to be passing now.

**Comment by TwitchBronBron** [comment]
Fixed by #112
